Set null protected areas in E and W to false

lizgzil opened the pull request:
Fixes #116 (description of why in issue description)

# Instructions for Reviewer

I found that all of England and Wales protected areas were either Null or True, so I think it is correct to set the Nulls to False, but perhaps I am missing something?

I ran 
```
python -i asf_heat_pump_suitability/pipeline/run_scripts/run_add_features.py --epc s3://asf-daps/lakehouse/processed/epc/old/deduplicated/processed_dedupl-0.parquet -y 2023 -q 4
```

to create `'s3://asf-heat-pump-suitability/outputs/2023Q4/20250204_2023_Q4_EPC_features.parquet'`

and then ran

```
python -i asf_heat_pump_suitability/pipeline/run_scripts/run_calculate_suitability.py --weights s3://asf-heat-pump-suitability/outputs/2023Q4/weights/20250102_2023_Q4_EPC_weights.parquet --features s3://asf-heat-pump-suitability/outputs/2023Q4/20250204_2023_Q4_EPC_features.parquet --gardens s3://asf-heat-pump-suitability/outputs/2023Q4/gardens/20250114_2023_Q4_EPC_garden_size_estimates_EWS_deduplicated.parquet -y 2023 -q 4
```

which saved to `s3://asf-heat-pump-suitability/outputs/2023Q4/suitability/20250206_2023_Q4_heat_pump_suitability_per_lsoa.csv`

lizgzil committed:
Set null protected areas in E and W to false

lizgzil committed:
Fix protected area logic to just fill nulls when there is lad conservation area data

crispy-wonton commented on asf_heat_pump_suitability/pipeline/run_scripts/run_add_features.py:
```diff
@@ -151,6 +151,13 @@ def parse_arguments() -> argparse.Namespace:
     )
 
+    epc_df = epc_df.with_columns(
+        pl.when(pl.col("lad_conservation_area_data_available_ew") == True)
```
```suggestion
        pl.when((pl.col("lad_conservation_area_data_available_ew")) & pl.col("COUNTRY").is_in(["England", "Wales"]))
```
I just pushed a change to filter the LAD conservation area data availability results to England and Wales LADs only. Therefore, I'm not sure if we actually need what I've added above, but I think it does add some robustness to the pipeline.

crispy-wonton committed:
update `protected_areas.py` to correctly fill nulls
- Scottish WHS nulls are filled with false only if they have a geometry, otherwise remain as nulls
- conservation area data availability df is filtered to only return results for England and Wales

crispy-wonton commented on asf_heat_pump_suitability/pipeline/prepare_features/protected_areas.py:
I have made 2 corrections in this file, which I noticed when looking at the outputs of your change:

1. For Scottish EPC records, we were filling all records with `False` for `in_protected_area` if the UPRN does not get joined to a World Heritage Site. However, not all records have valid coordinates, which is required to determine whether or not they are in a WHS. Scottish EPC records without coordinates should therefore receive `null`, and only those with coordinates should receive `False` if they are not joined to a WHS.
2. The `generate_df_conservation_area_data_availability` was producing a flag for data availability for all nations in the UK, but it was only using conservation area data from England and Wales to create the flag. Therefore, NI and Scotland were receiving `False` flags (meaning conservation area data is not available) for all their LADs - simply because we haven't actually input any cons area data for them. I have therefore filtered the output to England and Wales LADs only. This should prevent confusion and also increase robustness of the pipeline.

Note: I did notice when i was updating the `generate_df_conservation_area_data_availability` that a single Scottish LSOA (`S12000026`) did have LAD conservation area data available, while all others were False. I think since it appears to be a true Scottish LAD and therefore under Scottish governance, we can safely filter it out and apply the World Heritage Site rule only, but just flagging in case you have different thoughts.

Please do check my logic is correct. Let me know if you have questions.

crispy-wonton reviewed (commented):
Your change looks good and I've tested it works. I've added a suggestion to your code snippet which should be redundant with additional changes I have pushed, but increases robustness.

I noticed when I was looking at the outputs of your updated code that we have erroneously assigned `False` to some Scottish properties for `in_protected_area` flag, and that our `generate_df_conservation_area_data_availability` function was misleadingly retaining NI and Scotland LADs even though we do not use conservation area data for them. I have pushed fixes for these to this branch for you to take a look at, as well as a more detailed explanation in my comment on the file.

lizgzil committed:
Update asf_heat_pump_suitability/pipeline/run_scripts/run_add_features.py

Co-authored-by: crispy-wonton <104171770+crispy-wonton@users.noreply.github.com>

lizgzil commented:
Good spotting and suggestions @crispy-wonton - I think these look good. I'll start running it!
